select.go

```diff
@@ -199,15 +259,43 @@ func (e *ESql) convertComparisionExpr(expr *sqlparser.Expr, topLevel bool, paren
 	lhsStr = strings.Replace(lhsStr, "`", "", -1)
 
 	// extract rhs
-	// ? pass by pointer?
 	rhsStr, err := e.convertValExpr(&comparisonExpr.Right)
 	if err != nil {
 		return "", err
 	}
-
+	op := comparisonExpr.Operator
+	if not {
+		switch comparisonExpr.Operator {
+		case "=":
+			op = "!="
+		case "<":
+			op = ">="
+		case "<=":
+			op = ">"
+		case ">":
+			op = "<="
+		case ">=":
+			op = "<"
+		case "<>":
+			fallthrough
+		case "!=":
+			op = "="
+		case "in":
+			op = "not in"
+		case "not in":
+			op = "in"
+		case "like":
+			op = "not like"
+		case "not like":
+			op = "like"
+		default:
+			err := fmt.Errorf(`esql: %s operator not supported in comparison clause`, comparisonExpr.Operator)
+			return "", err
+		}
+	}
 	// generate dsl according to operator
 	var dsl string
-	switch comparisonExpr.Operator {
+	switch op {
```

Comment:
consider making it a function

Reply:
Do you mean a function like below?
`
func convertOp(op string) string {
    switch...
}
`